src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala

```diff
@@ -168,22 +168,26 @@ trait Checkpoints extends DeltaLogging {
    * @param cv The CheckpointVersion to compare against
    */
   protected def findLastCompleteCheckpoint(cv: CheckpointInstance): Option[CheckpointInstance] = {
-    var cur = math.max(cv.version, 0L)
-    while (cur >= 0) {
-      val checkpoints = store.listFrom(checkpointPrefix(logPath, math.max(0, cur - 1000)))
-          .map(_.getPath)
-          .filter(isCheckpointFile)
-          .map(CheckpointInstance(_))
-          .takeWhile(tv => (cur == 0 || tv.version <= cur) && tv.isEarlierThan(cv))
-          .toArray
-      val lastCheckpoint = getLatestCompleteCheckpointFromList(checkpoints, cv)
-      if (lastCheckpoint.isDefined) {
-        return lastCheckpoint
+    @annotation.tailrec
+    def loop(cur: Long): Option[CheckpointInstance] =
```

Comment:
I don't think making this call a tail recursion makes this code any more readable. In fact it adds friction to developers who don't know Scala in understanding the code base

Reply:
@brkyvz Sure, In that case, I am closing this PR. Thanks